include/uk/arch/ctx.h

```diff
@@ -194,6 +295,168 @@ void ukarch_ctx_init_entry2(struct ukarch_ctx *ctx,
  */
 void ukarch_ctx_switch(struct ukarch_ctx *store, struct ukarch_ctx *load);
 
+/**
+ * Initialize an auxiliary stack pointer. This must be always called the
+ * first time you create an auxiliary stack pointer.
+ *
+ * @param auxsp
+ *   The auxiliary stack pointer to initialize. Must point to the high end of
+ *  the auxiliary stack.
+ *
+ * NOTE: Auxiliary stack pointer must have UKARCH_AUXSP_ALIGN alignment.
+ *
+ */
+static inline void ukarch_auxsp_init(__uptr auxsp)
+{
+	struct ukarch_auxspcb *auxspcb_ptr;
+
+	UK_ASSERT(auxsp);
+	UK_ASSERT(IS_ALIGNED(auxsp, UKARCH_AUXSP_ALIGN));
+
+	auxspcb_ptr = (struct ukarch_auxspcb *)(auxsp - sizeof(*auxspcb_ptr));
+	auxspcb_ptr->curr_fp = auxsp - sizeof(*auxspcb_ptr);
+	UK_ASSERT(IS_ALIGNED(auxspcb_ptr->curr_fp, UKARCH_AUXSP_ALIGN));
+}
+
+/**
+ * Set the control block of the auxiliary stack pointer. At the moment,
+ * this function only copies the current frame pointer inside the auxstack area
+ * and the Unikraft system context.
+ *
+ * @param auxsp
+ *   The auxiliary stack pointer whose control block to set.
+ *  Must point to the high end of the auxiliary stack.
+ * @param auxspcb
+ *   The pointer to the control block to set for the auxstack pointer.
+ *
+ */
+static inline void ukarch_auxsp_set_cb(__uptr auxsp,
+				       struct ukarch_auxspcb *auxspcb)
+{
+	struct ukarch_auxspcb *auxspcb_ptr;
+
+	UK_ASSERT(auxsp);
+	UK_ASSERT(IS_ALIGNED(auxsp, UKARCH_AUXSP_ALIGN));
+
+	auxspcb_ptr = (struct ukarch_auxspcb *)(auxsp - sizeof(*auxspcb_ptr));
```

Comment:
The problem is uk_thread, struct lcpu and co. use the auxsp pointer `__uptr`, not the auxspcb pointer. The way I viewed this is that through `ukarch_auxsp_*` whoever may hold an auxsp pointer does not need to know of the auxspcb structure, unless they really want to.

Reply:
Following an offline discussion, it's been decided to move forward like so:
```C
static inline void ukarch_auxsp_init(__uptr auxsp);
static inline struct ukarch_auxspcb *ukarch_auxsp_get_cb(__uptr auxsp);
static inline void ukarch_auxspcb_set_uktlsp(struct ukarch_auxspcb *auxspcb, __uptr uktlsp);
static inline __uptr ukarch_auxspcb_get_uktlsp(struct ukarch_auxspcb *auxspcb);
static inline void ukarch_auxspcb_set_curr_fp(struct ukarch_auxspcb *auxspcb, __uptr curr_fp);
static inline __uptr ukarch_auxspcb_get_curr_fp(struct ukarch_auxspcb *auxspcb);
```
And no more getter or control block copy, as it would not be used at the moment anyway.